Services/Certificate/classes/class.ilCertificateAppEventListener.php

```diff
@@ -330,4 +340,33 @@ private function processEntry($type, $objectId, int $userId, ilCertificateTempla
 
         $this->certificateQueueRepository->addToQueue($entry);
     }
+
+    private function handleCompletedStudyProgramme()
+    {
+        $settings = new ilSetting('certificate');
+        $objectId = $this->parameters['prg_id'] ?? 0;
+        $userId = $this->parameters['usr_id'] ?? 0;
+        try {
+            $template = $this->templateRepository->fetchCurrentlyActiveCertificate($objectId);
+            if (true === $template->isCurrentlyActive()) {
```

Comment:
@klees Maybe `\ilCertificateAppEventListener::processEntry` could be called here instead?